content/news/buzz/2023q3-roadmap/contents.lr

```diff
@@ -0,0 +1,38 @@
+title: 2023Q3 Roadmap
+---
+author: Russell Keith-Magee
+---
+body:
+
+2023 is halfway done, and BeeWare continues to make slow but steady progress. As always, this roadmap should be read as a guide to what we aim to focus on over the coming quarter, rather than a hard commitment of features that will be made available on a specific deadline.
+
+Q2 progress
+-------------
+
+As with Q1, our focus in Q2 was on Toga testing. This has been a mammoth project; but it's finally nearing completion. All but 2 widgets (Canvas and DetailedList) have 100% branch coverage on macOS, iOS and GTK. Android and Windows. Along the way, we've identified and resolved lots of inconsistencies in widget APIs, and resolved countless bugs. We've also fully documented every widget, ported the test suite to Pytest, and added type annotations in the code (mostly for documentation purposes, rather than formal validation). We've also completed testing and coverage of a lot of the non-widget functionality that wasn’t strictly planned for this quarter - features like Images, Icons, Validators, application paths, and colors; plus a lot of tests (and bugfixes) in the core widget layout. We've also got the start of some tooling that can be used to test the Web backend.
+
+Unfortunately, we haven't been able to do any substantial Toga planning. Once Toga's code is at 100% testing coverage, we'll be in a much better place to develop those plans. We also haven't been able to add any native cross-platform hardware support.
+
+Q3 priorities
+--------------
+
+Our Q3 goals are:
+
+* **Finalize Toga's test suite**. By the end of Q3, we should have 100% branch coverage on every line of Toga's core, and on the macOS, Windows, GTK, Android and iOS backends.
+* **iOS Binary packaging** We've been made aware that Apple is rejecting Briefcase packaged iOS apps from the iOS App Store because of a package layout issue associated with binary modules. This is a very high priority to fix.
+* **Kotlin upgrades** The upcoming Android Studio release is switching to Kotlin as it's preferred language for new projects; we need to upgrade Chaquopy's tooling to use Kotlin. 
```

Comment:
Kotlin has been the preferred language for runtime code for a couple of years already, but now they're extending that to build.gradle files.

```suggestion
* **Kotlin upgrades** The upcoming Android Studio release is switching to Kotlin as its preferred language for new Gradle configuration files; we need to upgrade Chaquopy's tooling to support this. 
```